pkg/builder/build_test.go

```diff
@@ -121,6 +120,93 @@ var _ = Describe("application", func() {
 			Expect(err.Error()).To(ContainSubstring("expected error"))
 			Expect(instance).To(BeNil())
 		})
+
+		It("should scaffold a defaulting webhook if the type implements the Defaulter interface", func() {
+			By("creating a controller manager")
+			m, err := manager.New(cfg, manager.Options{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("registering the type in the Scheme")
+			builder := scheme.Builder{GroupVersion: schema.GroupVersion{Group: "foo.test.org", Version: "v1"}}
+			builder.Register(&TestDefaulter{}, &TestDefaulterList{})
+			err = builder.AddToScheme(m.GetScheme())
+			Expect(err).NotTo(HaveOccurred())
+
+			instance, err := ControllerManagedBy(m).
+				For(&TestDefaulter{}).
+				Owns(&appsv1.ReplicaSet{}).
+				Build(noop)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(instance).NotTo(BeNil())
+			svr := m.GetWebhookServer()
+			Expect(svr).NotTo(BeNil())
+
+			By("trying to register an existing mutating webhook path")
+			path := generateMutatePath(schema.GroupVersionKind{Group: "foo.test.org", Version: "v1", Kind: "TestDefaulter"})
+			Ω(func() { svr.Register(path, nil) }).Should(Panic())
```

Comment:
this is a bit of a weak test -- what if we add validation that handler is not nil, or something?  Maybe we should check that it responds to a valid admission request?

Reply:
Done.
PTAL